lib/tokens.ts

```diff
@@ -46,22 +47,48 @@ export function getTokensInfo(): TokenInfo[] {
  */
 export async function getTokenDetails(coingeckoId: string): Promise<TokenDetails> {
   const id = coingeckoId.toLowerCase()
-  return ALL_TOKENS.find(({ id: _id }) => _id === id)
+  const tokensRaw = await _getAllTokensData()
+  return tokensRaw.find(({ id: _id }) => _id === id)
 }
 
 function _getDescriptionFilePaths(): string[] {
   return fs.readdirSync(DESCRIPTIONS_DIR_PATH, 'utf-8')
 }
 
-function _getTokensRawInfo(): any[] {
-  const tokenJson = fs.readFileSync(TOKENS_PATH, 'utf-8')
-  const tokenData = JSON.parse(tokenJson)
+async function fetchWithBackoff(url) {
+  return backOff(
+    () => {
+      console.log(`Fetching ${url}`)
+      return fetch(url).then((res) => {
+        if (!res.ok) {
+          throw new Error(`Error fetching list ${url}: Error ${res.status}, ${res.statusText}`)
+        }
+
+        return res.json()
+      })
+    },
+    {
+      numOfAttempts: 50,
+      maxDelay: 36000000, // 10min
+      startingDelay: 5000, // 5s
+      retry: (e, attemptNum) => {
+        console.log(`Error fetching ${url}, attempt ${attemptNum}. Retrying soon...`)
+        return true
+      },
+    }
+  )
+}
+
+function _getTokensRawInfo(): Promise<any[]> {
+  // const tokenJson = fs.readFileSync(TOKENS_PATH, 'utf-8')
+  // const tokenData = JSON.parse(tokenJson)
 
-  return tokenData
+  // return tokenData
+  return fetchWithBackoff('https://raw.githubusercontent.com/cowprotocol/cow-fi/develop/data/tokens.json')
```

Comment:
We will need to point here to S3 URL 


I couldn't upload for now:
https://cowservices.slack.com/archives/C036PMLUPQF/p1688407496509999

Reply:
Great idea to load this externally 👍